backend/Watchdog.Core/Watchdog.Core.BLL/Services/DashboardService.cs

```diff
@@ -64,5 +65,15 @@ public async Task DeleteDashboardAsync(int dashboardId)
 
             await _context.SaveChangesAsync();
         }
+
+        public async Task<bool> IsDashboardNameValid(string dashboardName)
+        {
+            if (dashboardName.Length < 3 || dashboardName.Length > 50)
+            {
+                return false;
+            }
+
+            return !(await _context.Dashboards.ToListAsync()).Any(u => u.Name == dashboardName);
```

Comment:
Why do you need ToListAsync?

Reply:
I need to go through all the dashboards

Reply:
No, it's not what you need
First, of all, you need to check only dashboards that belong to the current organization. Second, you can use AnyAsync.
It is totally a bad decision to load ALL dashboards and after that do some manipulation with them

Reply:
Fix